src/java.base/share/classes/java/util/random/RandomGeneratorFactory.java

```diff
@@ -0,0 +1,524 @@
+/*
+ * Copyright (c) 2020, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package java.util.random;
+
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Method;
+import java.math.BigInteger;
+import java.security.AccessController;
+import java.security.PrivilegedActionException;
+import java.security.PrivilegedExceptionAction;
+import java.util.function.Function;
+import java.util.Map;
+import java.util.random.RandomGenerator.ArbitrarilyJumpableGenerator;
+import java.util.random.RandomGenerator.JumpableGenerator;
+import java.util.random.RandomGenerator.LeapableGenerator;
+import java.util.random.RandomGenerator.SplittableGenerator;
+import java.util.random.RandomGenerator.StreamableGenerator;
+import java.util.ServiceLoader;
+import java.util.ServiceLoader.Provider;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import jdk.internal.util.random.RandomSupport.RandomGeneratorProperty;
+
```

Comment:
Instead of calling a method properties to create a Map, it's usually far easier to use an annotation,
annotation values supports less runtime type so BigInteger is not supported but using a String instead should be OK.

Reply:
I kind of like the idea - not sure how expressive a BigInteger string is though. I might be able to express as <i, j, k>  BigInteger.ONE.shiftLeft(i).subtract(j).shiftLeft(k). Will ponder.

Reply:
Done

Reply:
Also added getDefault and getDefaultFactory to RandomGenerato.